### DISCUSSION TITLE
Unable to access RSS-Bridge feeds behind HTTP Basic Auth

### DISCUSSION THREAD

**Discussion by ghost** [FreshRSS/FreshRSS#5486] [Q&A] [answered]
**Describe the bug**

I have RSS-Bridge set up behind Nginx Proxy Manager's [HTTP Basic Authentication](https://docs.nginx.com/nginx/admin-guide/security-controls/configuring-http-basic-authentication/), because I only want certain users to have access to it. 

Typing in the Feed URL will not work since FreshRSS can't authenticate. You'll get an error:

```
cURL error 22: The requested URL returned error: 401 [https://rssbridge.mydomain.com/?action=display&amp;bridge=AssociatedPressNewsBridge&amp;context=Standard+Topics&amp;topic=apf-technology&amp;format=Html]
```

A workaround to be able to add the feed is to use:

```
https://username:password@rssbridge.mydomain.com/?action=display&amp;bridge=AssociatedPressNewsBridge&amp;context=Standard+Topics&amp;topic=apf-technology&amp;format=Html
```

This will add the feed to FreshRSS, then converts the URL back without `username:password@`. And the feed will not work. You can update the URL back with `username:password@`, but you'll just get a message saying: `Blast! This feed has encountered a problem. Please verify that it is always reachable then update it.`

I created a test login for it to put into [W3 Feed Validation Service](https://validator.w3.org/feed/). I used the URL:

```
https://testaccount:qwerty12345@rssbridge.mydomain.com/?action=display&amp;bridge=AssociatedPressNewsBridge&amp;context=Standard+Topics&amp;topic=apf-technology&amp;format=Html
```

It comes back as:

```
nonnumeric port: 'qwerty12345@rssbridge.mydomain.com' (InvalidURL; misconfigured server?)
```

I guess RSS can't parse that kind of URL.

It works if I have the user change `https://rssbridge.mydomain.com` to `http://my_private_ip:3000`, as it'll let FreshRSS directly access RSS-Bridge, but I do not want that. 1) They have to know my private IP.  2) They have to go out of their way to replace the beginning of the URL each time.

**To Reproduce**
1. Host RSS-Bridge with HTTP Basic Auth using Nginx Proxy Manager
2. Grab a feed from RSS-Bridge
3. Add feed to FreshRSS

**Expected behavior**
Expect to be able to grab feeds from RSS-Bridge even though it's behind HTTP Basic Authentication

**Environment information (please complete the following information):**
- Device: Desktop
- OS: Arch Linux
- Browser: Firefox 114.0.1
- FreshRSS version: 1.21.0 
- Database version: Built-in SQLite
- PHP version: 
- Installation type: Docker

**Comment by Frenzie** [comment, marked as answer]
Scroll down a bit (or click advanced during add feed). ;-)

Though filling it out automatically based on the URL would also be cool.

**Comment by ghost** [comment]
I can't believe I missed that... Now it works as expected. Thanks so much! Closing this issue.

Though if anyone is interested in:

> Though filling it out automatically based on the URL would also be cool.

Please open a new feature request issue :)
